feat: Deferred Revenue and Expense report with actual and upcoming postings

ruthra-kumar opened the pull request:
### Deferred Revenue and Expense Report
This report aim to provide a simplified view of all deferred income/expense in a certain period. It can help in identifying the actual and expected posting.

### Deferred Revenue
<img width="1552" alt="def_revenue" src="https://user-images.githubusercontent.com/3272205/145595902-5c57a12e-1559-4e61-a471-7aad7276cbd6.png">

### Deferred Expense
<img width="1552" alt="def_expense" src="https://user-images.githubusercontent.com/3272205/145595967-98d6e46e-d00f-46b7-a369-4fef1b02f0ca.png">

### Documentation
https://docs.erpnext.com/docs/v13/user/manual/en/accounts/deferred_revenue/expense_report

ruthra-kumar committed:
feat: Deferred Revenue and Expense report

 - show deferred revenue and expense with actual and expected postings
 - unit tests added

deepeshgarg007 commented on erpnext/accounts/report/deferred_revenue_and_expense/deferred_revenue_and_expense.py:
```diff
@@ -0,0 +1,436 @@
+		for item in self.uniq_items:
+			self.items.append(Deferred_Item(item, self, [x for x in items if x.item == item]))
+
+	def get_postings(self):
```
Seems like a dead function, not called from anywhere

ruthra-kumar replied:
removed

deepeshgarg007 commented on erpnext/accounts/report/deferred_revenue_and_expense/deferred_revenue_and_expense.js:
```diff
@@ -0,0 +1,114 @@
+		},
+		{
+			"fieldname" : "with_upcoming_postings",
+			"label": __("Show with upcoming revenue/expense"),
```
This also shows unbooked values for the previous months, may better to call this label as "Show expected revenue/expense" and this should be unchecked by default

ruthra-kumar replied:
Agreed. 
Along with this, I think it's better if it only shows the actual posting(bar graph) when unchecked. Thoughts?

deepeshgarg007 replied:
If someone wants to compare the expected and actual posting, how do they do that then?

ruthra-kumar replied:
renamed and 'unchecked' by default.

deepeshgarg007 commented:
Fix the linting issues

ruthra-kumar committed:
refactor: remove unused function 'get_postings'

ruthra-kumar committed:
refactor: rename checkbox and translations

 - rename checkbox and make it
 - added translation text for labels
 - fixes for linter error

deepeshgarg007 committed:
Merge branch 'develop' into deferred_report_revenue_and_expense

deepeshgarg007 commented:
@Mergifyio backport version-13-hotfix
